compiler/gradient_with_order.cc

```diff
@@ -111,7 +125,9 @@ void AddGradientNodesForTrainingWithOrders(Graph* graph, const std::vector<Order
         last_forward_map[orig_node] = node;
         for (const auto& p : Zip(node->outputs(), orig_node->outputs())) {
             Value* value = std::get<0>(p);
-            CHECK(staged.emplace(std::get<1>(p), value).second);
+            // TODO(mkusumoto): Can I safely remove this assertion?
+            // CHECK(staged.emplace(std::get<1>(p), value).second);
```

Comment:
I think we'd like to get this CHECK back eventually. I'm fine with this TODO, but how about showing an error message like

```
            if (!staged.emplace(std::get<1>(p), value).second) {
                std::cerr << "Forward recompute without forgetting the output: " << orig_node->ToString() << std::endl;
            }
```

Reply:
I modified the code so that warning message is shown.